system/assets/hypha-preview-image-file-field.js

```diff
@@ -0,0 +1,64 @@
+$(function() {
+	$('img[data-preview-for]').each(function () {
+		// get preview
+		var $preview = $(this);
+
+		// get input name
+		var inputName = $preview.data('preview-for');
+
+		// get input
+		var $input = $('input[name="' + inputName + '"][type="file"]');
+
+		// add button to remove the image file
+		var addRemoveImageBtn = function () {
+			$('<span class="hyphaRemoveButton" />')
+				.on('click', function (e) {
+					e.preventDefault();
+
+					// remove delete button
+					$(this).remove();
+
+					updateSrc(null);
+				})
+				.insertAfter($preview);
+		}
+
+		var toggle = function (hasSrc) {
+			if (hasSrc) {
+				$input.hide();
+				$('label[for="' + inputName + '"]').attr('for', inputName + '_disabled');
```

Comment:
Why is this needed?

Reply:
I ran into the problem that the label was still working on the hidden input. I just fixed this issue by disabling the input. :-)

Reply:
What does "working on" mean for a label anyway? The label is still shown, and associated with the same input, which is hidden. Doesn't sound (semantically) incorrect? Does this confuse the browser somehow? You could argue that the label should now associate with the preview or the remove button, but I'm not so sure this is possible or useful?

Reply:
It confusing for the user if a file upload dialog is triggered while that input field is hidden. I've added a comment. :-)

Reply:
Ah, I didn't realize that clicking the label triggers the input field, good point.